lib/richtext_controller.dart

```diff
@@ -0,0 +1,237 @@
+import 'dart:async';
+
+import 'package:enough_html_editor/enough_html_editor.dart';
+import 'package:flutter/material.dart' as ui;
+import 'package:flutter/material.dart';
+import 'package:rich_text_composer/views/widgets/rich_text_option_bottom_sheet.dart';
+
+import 'models/types.dart';
+
+class RichTextController {
+  HtmlEditorApi? htmlEditorApi;
+
+  final listSpecialTextStyleApply = ValueNotifier<Set<SpecialStyleType>>({});
+  final paragraphTypeApply = ValueNotifier<ParagraphType>(ParagraphType.alignLeft);
+  final dentTypeApply = ValueNotifier<DentType?>(null);
+  final orderListTypeApply = ValueNotifier<OrderListType?>(null);
+  final selectedTextColor = ValueNotifier<ui.Color>(ui.Colors.black);
+  final selectedTextBackgroundColor = ValueNotifier<ui.Color>(ui.Colors.white);
+  final headerStyleTypeApply = ValueNotifier<HeaderStyleType>(HeaderStyleType.normal);
+
+  final StreamController<bool> richTextStreamController =
+      StreamController<bool>();
+
+  bool isBoldStyleAppended = false;
+  bool isItalicStyleAppended = false;
+  bool isUnderlineAppended = false;
+  bool isStrikeThroughAppended = false;
+
+  Stream<bool> get richTextStream => richTextStreamController.stream;
+
+  void listenHtmlEditorApi() {
+    htmlEditorApi?.onFormatSettingsChanged = (formatSettings) {
+      if (formatSettings.isBold) {
+        isBoldStyleAppended = true;
+        listSpecialTextStyleApply.value.add(SpecialStyleType.bold);
+        listSpecialTextStyleApply.notifyListeners();
+      } else {
+        listSpecialTextStyleApply.value.remove(SpecialStyleType.bold);
+        isBoldStyleAppended = false;
+        listSpecialTextStyleApply.notifyListeners();
+      }
+
+      if (formatSettings.isItalic) {
+        isItalicStyleAppended = true;
+        listSpecialTextStyleApply.value.add(SpecialStyleType.italic);
+        listSpecialTextStyleApply.notifyListeners();
+      } else {
+        listSpecialTextStyleApply.value.remove(SpecialStyleType.italic);
+        isItalicStyleAppended = false;
+        listSpecialTextStyleApply.notifyListeners();
+      }
+
+      if (formatSettings.isUnderline) {
+        isUnderlineAppended = true;
+        listSpecialTextStyleApply.value.add(SpecialStyleType.underline);
+        listSpecialTextStyleApply.notifyListeners();
+      } else {
+        listSpecialTextStyleApply.value.remove(SpecialStyleType.underline);
+        isUnderlineAppended = false;
+        listSpecialTextStyleApply.notifyListeners();
+      }
+
+      if (formatSettings.isStrikeThrough) {
+        isStrikeThroughAppended = true;
+        listSpecialTextStyleApply.value.add(SpecialStyleType.strikeThrough);
+        listSpecialTextStyleApply.notifyListeners();
+      } else {
+        listSpecialTextStyleApply.value.remove(SpecialStyleType.strikeThrough);
+        isStrikeThroughAppended = false;
+        listSpecialTextStyleApply.notifyListeners();
+      }
+    };
+  }
+
+  selectTextStyleType(SpecialStyleType richTextStyleType) {
```

Comment:
If there is no function return, you should use the `void` keyword before the function name.